src/librustc_session/config.rs

```diff
@@ -1698,11 +1698,29 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         }
     }
 
-    if cg.profile_generate.enabled() && cg.profile_use.is_some() {
-        early_error(
-            error_format,
-            "options `-C profile-generate` and `-C profile-use` are exclusive",
-        );
+    if debugging_opts.instrument_coverage && !cg.profile_generate.enabled() {
+        // `-Z instrument-coverage implies `-C profile-generate` (to generate additional LLVM
+        // runtime dependencies) and `-C link-dead-code` (so unexecuted code is still counted
+        // as zero, rather than be optimized out). Note that instrumenting dead code can be
```

Comment:
Is there a (future) way of accomplishing this without linking dead code? Is that what the "unreachable counters" FIXME is for?

Reply:
Not unreachable, but yes, I was thinking that I might be able to use a "zero counter" for this (see [Counters](https://llvm.org/docs/CoverageMappingFormat.html#counter)).

I'll add a FIXME to investigate it.

Reply:
FIXME added for now.